ehrapy/plot/_sankey.py

```diff
@@ -0,0 +1,267 @@
+
+    return pd.DataFrame({"source": sources, "target": targets, "value": values})
+
+
+@choose_hv_backend()
+def sankey_diagram_time(
+    edata: EHRData,
+    *,
+    var_name: str,
+    layer: str,
+    state_labels: dict[int, str] | None = None,
+    node_width: int | float = 20,
+    node_padding: int | float = 10,
+    node_color: str | None = None,
+    edge_color: str | None = None,
+    label_position: str | None = "right",
+    show_values: bool = True,
+    title: str | None = None,
+    width: int | None = 600,
+    height: int | None = 400,
+    **kwargs,
+) -> hv.Sankey:
+    """Create a Sankey diagram showing patient state transitions over time.
+
+    Each node represents a state at a specific time point, and flows show the
+    number of patients transitioning between states.
+    Visualizes how patients transition between different states
+    (e.g. disease severity, treatment status) across consecutive time points.
+
+    Args:
+        edata: Central data object.
+        var_name: Variable name from  `edata.var_names` to visualize
+        layer: Name of the layer in `edata.layers` containing the feature data to visualize.
+        state_labels: Mapping from numeric state values to readable labels.
+                    If None, state values will be displayed as strings of their numeric codes (e.g., "0", "1", "2").
+        node_width : Width of the nodes in the Sankey diagram.
+        node_padding : Padding between nodes in the Sankey diagram.
+        node_color : Color of the nodes. If None, default coloring is used.
+        edge_color: Color of the edges. If None, default coloring is used.
+        label_position : Position of the labels on the nodes. Options are 'left', 'right', 'outer', or 'inner'.
+        show_values : Whether to display the values on the edges.
+        title : Title of the Sankey diagram.
+        width : Width of the Sankey diagram.
+        height : Height of the Sankey diagram.
+        **kwargs: Additional styling options passed to :class:`holoviews.element.sankey.Sankey`.
+
+    Examples:
+        >>> import ehrapy as ep
+        >>> import ehrdata as ed
+        >>> edata = ed.dt.ehrdata_blobs(base_timepoints=5, n_variables=1, n_observations=5, random_state=59)
+        >>> edata.layers["tem_data"] = edata.layers["tem_data"].astype(int)
+        >>> state_labels = {-2: "no", -3: "mild", -4: "moderate", -5: "severe", -6: "critical"}
+        >>> ep.pl.sankey_diagram_time(
+        ...     edata,
+        ...     var_name="feature_0",
+        ...     layer="tem_data",
+        ...     state_labels=state_labels,
+        ... )
+
+        .. image:: /_static/docstring_previews/sankey_time.png
+    """
+    if var_name not in edata.var_names:
+        raise KeyError(f"{var_name} not found in edata.var_names.")
+    if layer not in edata.layers:
+        raise KeyError(f"{layer} not found in edata.layers.")
+
+    flare_data = edata[:, edata.var_names.isin([var_name]), :].layers[layer][:, 0, :]
```

Comment:
[This comment](https://github.com/theislab/ehrapy/pull/989#pullrequestreview-3588488317) remains, I still don't think multiple columns should be allowed to pass but the second case still behaves in the same way - something like this change will stop this from unexpectedly working I think

```suggestion
    flare_data = edata[:, edata.var_names == var_name), :].layers[layer][:, 0, :]
```